analyses/simulate_transmission/epidemics_odin_interop.qmd

```diff
@@ -0,0 +1,133 @@
+---
+title: "How to use _epidemics_ classes with _odin_-generated models"
+format: 
+  html:
+    code-link: true
+editor: source
+editor_options: 
+  chunk_output_type: console
+date: last-modified
+toc: true
+toc_float: true
+---
+
+## Ingredients
+
+This guide shows how to use some of the convenient features of _epidemics_, especially the classes that describe populations, interventions, and vaccination regimes, with models that are generated by _odin_.
+
+This guide shows this interoperability by implementing the default model from _epidemics_ in _odin_.
+
+```{r}
+library(epidemics)
+library(socialmixr)
+library(odin)
+library(data.table)
+library(ggplot2)
+```
+
+## Steps in code
+
+### Prepare population initial conditions
+
+```{r}
+# get social contacts data
+polymod <- socialmixr::polymod
+contact_data <- socialmixr::contact_matrix(
+  polymod,
+  countries = "United Kingdom",
+  age.limits = c(0, 20, 40),
+  symmetric = TRUE
+)
+
+# assume arbitrary populations for each demo group
+demography_vector <- contact_data$demography$population
+
+# prepare contact matrix, divide by leading eigenvalue and rowwise by popsize
+contact_matrix <- t(contact_data[["matrix"]])
+contact_matrix <- (contact_matrix / max(Re(eigen(contact_matrix)$values))) /
+  demography_vector
+
+# initial conditions defined as proportions and converted to absolute values
+initial_condtions <- matrix(
+  c(S = 1 - 1e-6, I = 1e-6, R = 0),
+  nrow = length(demography_vector),
+  ncol = 3L, byrow = TRUE
+) * demography_vector
+```
+
+### Define an epidemic model in _odin_
+
+This is an SIR model that omits the 'exposed' compartment for this demonstration.
+
+```{r}
+# define SIR model
+sir <- odin::odin({
+  # NOTE: crude time-dependence of transmission rate
+  # similar to a `<rate_intervention>`
+  beta_ <- if (t > intervention_interval[1] && t < intervention_interval[2]) beta * (1.0 - reduction) else beta
```

Comment:
```suggestion
  beta_t[] <- if (t > intervention_interval[1] && t < intervention_interval[2]) beta * (1.0 - reduction[i]) else beta*(1 + 0*reduction[i])
```
Note need [i] indexing on RHS and odin doesn't allow multi-line if statements. But could pre-define up to X interventions if need to be able to handle multiple interventions.